latest/index.bs

```diff
@@ -192,7 +189,10 @@ keys as specified below for discovering certain types of data, especially images
     - Units for "space" axes: 'angstrom', 'attometer', 'centimeter', 'decimeter', 'exameter', 'femtometer', 'foot', 'gigameter', 'hectometer', 'inch', 'kilometer', 'megameter', 'meter', 'micrometer', 'mile', 'millimeter', 'nanometer', 'parsec', 'petameter', 'picometer', 'terameter', 'yard', 'yoctometer', 'yottameter', 'zeptometer', 'zettameter'
     - Units for "time" axes: 'attosecond', 'centisecond', 'day', 'decisecond', 'exasecond', 'femtosecond', 'gigasecond', 'hectosecond', 'hour', 'kilosecond', 'megasecond', 'microsecond', 'millisecond', 'minute', 'nanosecond', 'petasecond', 'picosecond', 'second', 'terasecond', 'yoctosecond', 'yottasecond', 'zeptosecond', 'zettasecond'
 
-If part of [[#multiscale-md]], the length of "axes" MUST be equal to the number of dimensions of the arrays that contain the image data.
+The "axes" are used as part of [[#multiscale-md]]. The length of "axes" MUST be equal to the number of dimensions of the arrays that contain the image data.
+
+The "dimension_names" attribute in the `zarr.json` of the Zarr array of a multiscale level MUST match the names in the "axes" metadata.
```

Comment:
"dimension_names" are redundant in an OME-Zarr multiscale image, so must they be mandatory? Perhaps this restriction could be relaxed to something like:
 > If the "dimension_names" attribute is specified in the `zarr.json` of the Zarr array of a multilscale level, it SHOULD match the names in the "axes" metadata.

This will enable arrays with undesirable/non-descriptive/missing dimension names to be used in an OME-Zarr hierarchy without any array metadata changes.

Reply:
I am open to discuss this. Weakening this restriction could cause conflicts between the array metadata and the OME-Zarr metadata. We would need to define a precedence order. 
I wonder what the circumstances would be that you can add the OME-Zarr metadata on the group level, but cannot adjust the array metadata to match the "dimensions_names" attribute?

Reply:
I can always work around this, so it is not essential.

I have legacy multiscale arrays that have been converted from NetCDF that I would *prefer* to keep immutable and a one-to-one mapping to their source. I want to slap OME-Zarr on top with sensible `axes` names (e.g. `z`, `y`, `x`). However, the `dimension_names` of the underlying arrays may encode other information and be inconsistent between scales (e.g. `segmented_x_1.23_um`).

Reply:
The 0.4 spec has many restrictions on the underlying arrays (consecutively numbered groups, the order of dimensions, nested directory layout, etc.) that have since been addressed by this RFC and [RFC-3](https://github.com/ome/ngff/pull/239). As far as I can see[^1], the `dimension_names` restriction introduced in this RFC is the only remaining restriction that could make arrays incompatible with OME-Zarr metadata (aside from requiring the length of `axes` to match the number of dimensions of the arrays, which is necessary).

[^1]: I've only read the spec for `multiscales` and dependent metadata

Reply:
@d-v-b What do you think about this? I believe you advocated for strictly keeping `dimension_names` in sync.

Reply:
I think they should be kept in sync, because the alternative is confusing -- how should clients interpret variance between `axes` and `dimension_names`? And if clients are supposed to just ignore `dimension_names`, then why did we add it to the zarr v3 spec in the first place?